modules/programs/rbrowser/default.nix

```diff
@@ -1,20 +1,60 @@
-{ mode, config, pkgs, lib, ... }:
+{ mode, config, home-manager, lib, pkgs, ... }:
 
 with lib;
 let
   cfg = config.soxin.programs.rbrowser;
 
-  browserSubmodule = { name, ... }: {
-    options = {
-      enable = mkOption {
-        default = true;
-        type = types.bool;
-        description = ''
-          Whether to enable the ${name} integration with rbrowser.
-        '';
+  browserSubmodule =
+    let
+      getFromDagName = dagName: index:
+        let
+          hasAt = lib.stringAsChars (x: if x == "@" then x else "") dagName == "@";
+          name_profile = builtins.head (builtins.tail (builtins.split "^(.*)@(.*)$" dagName));
+        in
+        assert lib.asserts.assertMsg hasAt ''The dagName must be of the format browser@profile, got "${dagName}"!'';
+        assert lib.asserts.assertMsg (builtins.length name_profile == 2) "The dagName must be of the format browser@profile.";
+        builtins.elemAt name_profile index;
+    in
+    { dagName, ... }:
+    {
+      options = {
+        enable = mkOption {
```

Comment:
Do we still need this option? The user could simply remove the entry to disable it.

Reply:
Probably not. I'll get rid of it.